src/language_server_protocol.rs

```diff
@@ -1824,6 +1824,32 @@ impl LanguageClient {
         Ok(())
     }
 
+    pub fn window_showMessageRequest(&self, params: &Value) -> Fallible<Value> {
+        info!("Begin {}", lsp::request::ShowMessageRequest::METHOD);
+        let msg_params: ShowMessageRequestParams = params.clone().to_lsp()?;
+        let msg_actions = msg_params.actions.unwrap_or_default();
+        let n_actions = msg_actions.len();
+        let actions: Vec<Value> = msg_actions
+            .into_iter()
+            .enumerate()
+            .map(|(i, item)| Value::String(format!("{}) {}", i + 1, item.title)))
+            .collect();
+        let mut options: Vec<Value> = Vec::with_capacity(actions.len() + 1);
+        options.push(Value::String(msg_params.message));
+        options.extend(actions);
+
+        let mut v = Value::Null;
+        let result: Option<i64> = self.vim()?.rpcclient.call("s:inputlist", options)?;
+        if let Some(answer) = result {
+            if answer > 0 && answer < (n_actions - 1) as i64 {
+                let raw_actions: Vec<Value> = try_get("actions", &params)?.unwrap_or_default();
```

Comment:
Parse action is not necessary since there is already deserialized instance before hand. Around L1833, you can use `iter()`  instead of `into_iter()` so you can continue use the instance here.

Reply:
Yeah I thought the struct in lsp-type wouldn't contain everything that's required, but I just checked the spec and it does. Will fix the code.

Reply:
Question: what's the easiest way to convert ``MessageActionItem`` into ``Value``? 🤔

Reply:
`serde_json::to_value`